README.md

```diff
@@ -18,6 +18,62 @@ ArmoniK.Core provides services for submitting computational tasks, keeping track
 
 More detailed information on the inner working of ArmoniK.Core is available [here](https://aneoconsulting.github.io/ArmoniK.Core/).
 
+## Development process
+
+This repository follows trunk based development so all features will be integrated into main, in multiple small steps.
+Breaking changes will be limited as much as possible or gated behind feature flags.
+If breaking changes should happen, they will be documented in the releases.
+Therefore, features in development or testing that cannot fit within a branch and need to be integrated into main will be marked as preview so that users know that these features are still in development and subject to changes.
+
+**General availability:** The APIs, plugins and features that are generally available are highly curated and for which support is available.
+**Preview:** The APIs, plugins and features that are available on main but still in development. Their API should be stable enough for advanced users who want to test them before GA. Breaking changes should be minimal and will be notified in the release notes. Any feedback or issue encountered with these features are welcome !
+**Beta (Internal preview):** The APIs, plugins and features in active development. They are not stable enough and breaking changes may occur without any notice. Breaking changes will not always be documented. They can be used by the community for testing purposes. Before to be moved to the "Preview" stage, issues might be closed with the minimal messages "Under Active Development".
```

Comment:
```suggestion
**Beta (Internal preview):** The APIs, plugins and features in active development. They are not stable enough and breaking changes may occur without any notice. Breaking changes will not always be documented. They can be used by the community for testing purposes. Before being moved to the "Preview" stage, issues might be closed with the minimal messages "Under Active Development".
```

Not exactly sure about the meaning behind the last sentence

Reply:
We'll close issues about Beta features without notice